src/components/Publish/_utils.ts

```diff
@@ -506,16 +507,53 @@ export async function verifyRawServiceCredential(
       }
     }
     if (response?.status === 201) {
-      const serviceOffering = parsedServiceCredential.verifiableCredential.find(
-        (credential) =>
-          credential?.credentialSubject?.type === 'gx:ServiceOffering'
-      )
-      const credentialId = serviceOffering?.credentialSubject?.id
-
-      return {
-        verified: true,
-        complianceApiVersion,
-        idMatch: did && did?.toLowerCase() === credentialId?.toLowerCase()
+      const serviceOfferings =
+        parsedServiceCredential.verifiableCredential.filter(
+          (credential) =>
+            credential?.credentialSubject?.type === 'gx:ServiceOffering'
+        )
+      if (serviceOfferings.length === 1) {
+        return {
+          verified: true,
+          complianceApiVersion,
+          idMatch:
+            did &&
+            did?.toLowerCase() ===
+              serviceOfferings?.credentialSubject?.id.toLowerCase()
+        }
+      } else {
+        const dependsOnIds = serviceOfferings
+          .filter((service) => service?.credentialSubject?.['gx:dependsOn'])
+          .flatMap((service) => service?.credentialSubject?.['gx:dependsOn'])
+          .flatMap((dependsOn) => dependsOn?.id)
+
+        const rootService = serviceOfferings
+          .filter(
+            (service) => !dependsOnIds.includes(service?.credentialSubject?.id)
+          )
+          .map((service) => service?.credentialSubject?.id)
+
+        if (rootService.length > 1) {
+          return {
+            verified: true,
+            complianceApiVersion,
+            idMatch: rootService?.includes(did || did.toLowerCase()),
+            isIdMatchVerifiable: 'Too many root services'
+          }
+        } else if (rootService.length === 0) {
```

Comment:
suggestion/nit-pick(non-blocking): I'd prefer `.length < 1` for easier readability and understanding of last statement